modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -72,7 +85,7 @@
     @Override
     public IngestDocument execute(IngestDocument document) {
         IngestScript.Factory factory = scriptService.compile(script, IngestScript.CONTEXT);
-        factory.newInstance(script.getParams()).execute(document.getSourceAndMetadata());
+        factory.newInstance(script.getParams()).execute(new DeprecationMap(document.getSourceAndMetadata(), DEPRECATIONS));
```

Comment:
I noticed that `DeprecationMap` uses `deprecated` instead of `deprecatedAndMaybeLog`. Is this something we might want to change?

Reply:
That's a good point.  @rjernst What are your thoughts here?

Reply:
Done.